### ISSUE TITLE
avg_log2FC Values from FindMarkers Seem Inappropriately Low

### ISSUE THREAD

**Issue by aideenoneill** [satijalab/seurat#7749]
Dear Sajitalab or anyone who can help,

I'd be extremely grateful for help on the following - I cannot see a fix from previous issues. I don't have much coding/ Seurat experience.

I have run FindMarkers on an integrated, SCTransform'd object, with the objective of generating LFC values comparing Control vs Leukaemia.  In my FindMarkers code, I specified "RNA Assay" and "data" as the slot.  Prior to having run integration, I had run NormalizeData(..., normalization.method = "LogNormalize") on the underlying datsets, so the RNA Assay does contain normalized values.  

My issue is that the avg_log2FC values produced by FindMarkers are almost all <1 and seem inappropriately low compared to what is suggested by looking at the VlnPlots across the two conditions.  Eg, gene x has an expression level of ~1.3 at the midpoint of the violin for Control, versus negligible for Leukaemia, therefore I'd expect a LFC of well above 1.  However the log2FC is only 0.8.  

As a separate reference point, previous analysis performed by someone else on the same datasets (but not using SCT or integration) resulted in virtually identical violin plot values but much higher avg_log2FCs for almost all genes through FindMarkers.   I've studied that code but cannot see any issues.

I'm concerned that there is an error in my FindMarkers code whereby I'm not comparing the things that I think I am (eg, comparing to other clusters, versus comparing Coontrol vs. Leukaemia.  OR whereby, I'm taking my FindMarkers data from the wrong slot.  I've pasted in the relevant section of my code.

Thank you for any help you can give me.

#Run FindMarkers
Idents(sobj_qc_integrated) <- sobj_qc_integrated$Cell_Type 
treat.celltype.de <- list()

for (type in unique(Idents(sobj_qc_integrated))) {
  treat.celltype.de[[type]] <- FindMarkers(object = subset(sobj_qc_integrated, ident = type),
                                           group.by = "orig.ident", 
                                           ident.1 = "bm_str_control", ident.2 = "bm_str_leukaemia", 
                                           test.use = "MAST",
                                           slot = "data",
                                           assay = "RNA",
                                           logfc.threshold = 0.25,
                                           min.pct  = 0.1,
                                           only.pos = FALSE,
                                           base = 2
                                           )
  }

treat.celltype.de <- lapply(treat.celltype.de, function(x) rownames_to_column(x, "Gene"))
write_xlsx(treat.celltype.de, "treat.celltype.de.xls")

**Comment by saketkc** [comment]
Hi @aideenoneill, what version of Seurat are you using? I would recommend installing the latest develop branch where we made some fixes to the FindMarker function. That said it would be useful if you could paste your entire code and the plots.

```r
# Enter commands in R (or R studio, if installed)
# Install the remotes package
install.packages('remotes')
remotes::install_github(repo = 'satijalab/seurat', ref = 'develop')
library(Seurat)
```

**Comment by saketkc** [comment]
This is addressed in Seurat 4.4.0 and Seurat 5.0.0 revises how fold changes were calculated.
